### PR TITLE
arrow-core: speedup 'Arb.string()' related slow tests

### PR DESCRIPTION
**Pull request by abendt**
Some of the tests in arrow-core run slow on the non jvm platforms (https://github.com/arrow-kt/arrow/issues/2908)

One reason seems to be the use of `Arb.string()` especially when used inside arbs that generate collections (map, list, nonemptylist) 

See
* https://github.com/arrow-kt/arrow/pull/2999#issuecomment-1487382990
* https://github.com/arrow-kt/arrow/pull/2999#issuecomment-1488061474

This MR replaces `Arb.string()` with `Arb.int()` in some tests to speed up the execution time

### CODE REVIEW

**Comment by abendt** [comment]
Before https://scans.gradle.com/s/ogbwpswsaig4q/tests/overview

image

After https://scans.gradle.com/s/i6grm52lc7v6u/tests/overview

image

File: arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/IorSpec.kt [hunk 1]
```diff
@@ -32,11 +33,11 @@ class IorSpec :
     }
 
     "Concurrent - arrow.ior bind" {
-      checkAll(Arb.list(Arb.string()).filter(List<String>::isNotEmpty)) { strs ->
+      checkAll(Arb.list(Arb.int(), 1 .. 100)) { xs ->
```
Comment (nomisRev) [inline arrow-libs/core/arrow-core/src/commonTest/kotlin/arrow/core/continuations/IorSpec.kt:36]:
```suggestion
      checkAll(Arb.nonEmptyList(Arb.int())) { xs ->
```

**Review by nomisRev — APPROVED** [review]
Looks awesome @abendt. Just two small nits (probably missing proper imports, so suggestions is going to break).

**Comment by abendt** [comment]
> A bit related, but also unrelated to this. I am a bit curious if running the tests on latest JDK would increase build times as well. IIRC only building the binaries in the `publish` step need to run in JDK 11? Related tweet, https://twitter.com/EsaFirm/status/1641083869083602944?s=20

Sure, that could prohably improve build times. More ideas:
* configuration one of the gradle plugins is really slow. This makes it very painful to run tests in IntelliJ. You can also see that in the build scan: https://scans.gradle.com/s/i6grm52lc7v6u/performance/configuration. I commented out the plugin locally (https://github.com/arrow-kt/arrow/blob/main/build.gradle.kts#L34)
* In other projects i have used the gradle build cache with good results

**Comment by serras** [comment]
> Sure, that could prohably improve build times. More ideas:
> 
> * gradle configuration takes a long time because of one of the gradle plugins. This makes it very painful to run tests in IntelliJ. You can also see that in the build scan: https://scans.gradle.com/s/i6grm52lc7v6u/performance/configuration. I had to comment out the plugin locally (https://github.com/arrow-kt/arrow/blob/main/build.gradle.kts#L34)

We actually are working on improving that in https://github.com/arrow-kt/arrow/pull/2988, although we may fix it once and for all in the Gradle plug-in we use https://github.com/arrow-kt/arrow-gradle-config/pull/69

**Review by franciscodr — APPROVED** [review]
Thanks for helping us improve the build times, @abendt I appreciate the time you're putting on this!

**Pull request merged**